presto-array/src/main/java/com/facebook/presto/array/Arrays.java

```diff
@@ -108,7 +110,7 @@ else if (buffer.length < capacity) {
 
     public static byte[] ensureCapacity(byte[] buffer, int capacity, int estimatedMaxCapacity, ExpansionFactor expansionFactor, ExpansionOption expansionOption, ArrayAllocator allocator)
     {
-        int newCapacity = max(capacity, min((int) (capacity * expansionFactor.expansionFactor), estimatedMaxCapacity));
+        int newCapacity = (int) max(capacity, min((int) (capacity * expansionFactor.expansionFactor), estimatedMaxCapacity * graceFactorFordMaxCapacity));
```

Comment:
This is a generic method that can be used in many places. However, the commit says that the change applies only to one specific use case. I'd expect the caller to apply this new factor when computing estimatedMaxCapacity.

- use Math.toIntExact instead of (int)

Reply:
> This is a generic method that can be used in many places. However, the commit says that the change applies only to one specific use case. I'd expect the caller to apply this new factor when computing estimatedMaxCapacity.

Moved the application of this new factor to `setupDecodedBlockAndMapPositions()` where the estimatedMaxCapacity is calculated.

> * use Math.toIntExact instead of (int)

It's actually casting `double` to `int`. `toIntExact(long)` only takes `long`.